File: src/main/java/com/logics/logics/repositories/GameRoomRepository.java
package com.logics.logics.repositories;

import com.logics.logics.entities.GameRoom;
import org.springframework.data.repository.reactive.ReactiveCrudRepository;
import reactor.core.publisher.Flux;
import reactor.core.publisher.Mono;

public interface GameRoomRepository extends ReactiveCrudRepository<GameRoom, String> {
  Flux<GameRoom> findByStatus(GameRoom.GameRoomStatus status);
  Mono<GameRoom> findByIdAndStatus(String id, GameRoom.GameRoomStatus status);
}

File: src/main/java/com/logics/logics/repositories/GameRoomRepository.kt
package com.logics.logics.repositories

import com.logics.logics.entities.GameRoom
import org.springframework.data.repository.reactive.ReactiveCrudRepository
import reactor.core.publisher.Flux
import reactor.core.publisher.Mono

interface GameRoomRepository : ReactiveCrudRepository<GameRoom, Long> {
    fun findByStatus(status: GameRoom.GameRoomStatus): Flux<GameRoom>
    fun findByIdAndStatus(id: Long, status: GameRoom.GameRoomStatus): Mono<GameRoom>
}